src/routes/api/agents/[agentId]/conversations/+server.ts

```diff
@@ -1,40 +1,80 @@
 import { json, type RequestHandler } from "@sveltejs/kit"
+import { logger } from "@vestfoldfylke/loglady"
 import { createAgent, getDBAgent } from "$lib/server/agents/agents.js"
-import { deleteDBConversation, getDBConversations, insertDBConversation, updateDBConversation } from "$lib/server/agents/conversations.js"
+import { deleteDBConversation, getDBAgentUserConversations, insertDBConversation, updateDBConversation } from "$lib/server/agents/conversations.js"
+import { getUserInputTextFromPrompt } from "$lib/server/agents/message"
+import { canPromptAgent, canViewConversation } from "$lib/server/auth/authorization"
+import { HTTPError } from "$lib/server/middleware/http-error"
+import { httpRequestMiddleware, type MiddlewareNextFunction } from "$lib/server/middleware/http-request"
 import { responseStream } from "$lib/streaming"
+import type { GetAgentConversationsResponse } from "$lib/types/api-responses"
 import { ConversationRequest } from "$lib/types/requests"
 
-// OBS OBS Kan hende vi bare skal ha dette endepunktet - og dersom man ikke sender med en conversationId så oppretter vi en ny conversation, hvis ikke fortsetter vi den eksisterende (ja, kan fortsatt kanskje hende det)
+const getAgentUserConversations: MiddlewareNextFunction = async ({ requestEvent, user }) => {
+	if (!requestEvent.params.agentId) {
+		throw new HTTPError(400, "agentId is required")
+	}
+	if (!user.userId) {
+		throw new HTTPError(400, "userId is required")
+	}
+	const agentUserConversations = await getDBAgentUserConversations(requestEvent.params.agentId, user.userId)
 
-export const GET: RequestHandler = async ({ params }): Promise<Response> => {
-	// Da spør vi DB om å hente conversations som påkaller har tilgang på i denne assistenten
-	if (!params.agentId) {
-		throw new Error("agentId is required")
+	const authorizedConversations = agentUserConversations.filter((conversation) => canViewConversation(user, conversation))
+	const unauthorizedConversations = agentUserConversations.filter((conversation) => !canViewConversation(user, conversation))
+	if (unauthorizedConversations.length > 0) {
+		// This should not happen as getDBAgents filters based on user access
+		logger.warn(
+			`User: {userId} got {count} conversations they are not authorized to view from db query. Filtered them out, but take a look at _ids {@ids}`,
```

Comment:
Didn't `biome` nag about using template liter strings when not needed? 🤔

```suggestion
			"User: {userId} got {count} conversations they are not authorized to view from db query. Filtered them out, but take a look at _ids {@ids}",
```